src/test/java/org/broadinstitute/hellbender/utils/mcmc/AdaptiveMetropolisSamplerUnitTest.java

```diff
@@ -0,0 +1,106 @@
+ */
+public class AdaptiveMetropolisSamplerUnitTest {
+    private static final double BETA_MIN = 0;
+    private static final double BETA_MAX = 1;
+    private static final double BETA_START = 0.5;
+    private static final double INITIAL_STEP_SIZE = 0.1;
+    private static final EmptyData EMPTY_DATA = new EmptyData();
+    private static final int NUM_BURN_IN_STEPS = 200;
+    private static final int NUM_SAMPLES = 500;
+
+
+    // factory for sampling the distribution Beta(a, b) defined on [0,1]
+    final static GibbsSampler<DoubleState, EmptyData> betaGibbsSampler(final double a, final double b) {
+        final  Sampler<Double, DoubleState, EmptyData> betaSampler =
+                new AdaptiveMetropolisSampler<DoubleState, EmptyData>(INITIAL_STEP_SIZE, BETA_MIN, BETA_MAX) {
+                @Override
+                protected double proposalLogLikelihood(Double proposal, DoubleState state, EmptyData data) {
+                    return (a - 1) * Math.log(proposal) + (b - 1) * Math.log(1 - proposal);
+                }
+
+                @Override
+                protected double currentValue(final DoubleState state) {
+                    return state.value();
+                }
+            };
+        ParameterizedModel<DoubleState, EmptyData> betaModel =
+                new ParameterizedModel.GibbsBuilder<>(new DoubleState(BETA_START), EMPTY_DATA, DoubleState.class)
+                .addParameterSampler(DoubleState.PARAMETER_NAME, betaSampler, Double.class).build();
+
+        return new GibbsSampler<>(NUM_BURN_IN_STEPS + NUM_SAMPLES, betaModel);
+        }
+
+
+    // factory for sampling Gaussian with mean mu and precision lambda
+    final static GibbsSampler<DoubleState, EmptyData> gaussianGibbsSampler(final double mu, final double lambda) {
+        final  Sampler<Double, DoubleState, EmptyData> gaussianSampler =
+                new AdaptiveMetropolisSampler<DoubleState, EmptyData>(INITIAL_STEP_SIZE) {
+                    @Override
+                    protected double proposalLogLikelihood(Double proposal, DoubleState state, EmptyData data) {
+                        final double x = state.value();
+                        return -(lambda/2)*(proposal-mu)*(proposal-mu);
+                    }
+
+                    @Override
+                    protected double currentValue(final DoubleState state) {
+                        return state.value();
+                    }
+                };
+
+        final ParameterizedModel<DoubleState, EmptyData> gaussianModel =
+                new ParameterizedModel.GibbsBuilder<>(new DoubleState(BETA_START), EMPTY_DATA, DoubleState.class)
+                .addParameterSampler(DoubleState.PARAMETER_NAME, gaussianSampler, Double.class).build();
+
+        return new GibbsSampler<>(NUM_BURN_IN_STEPS + NUM_SAMPLES, gaussianModel);
+    }
+
+
+    @Test
+    public void testBeta() throws Exception {
+        for (final double a : Arrays.asList(10, 20, 30)) {
```

Comment:
could also use DataProviders here (although it wouldn't save you as many lines of code in this case, could make things a bit more readable---will leave it up to you)